lib/components/SchematicViewer.tsx

```diff
@@ -90,21 +107,28 @@ export const SchematicViewer = ({
     }
   }, [svgString])
 
+  const handleEditEvent = (event: ManualEditEvent) => {
+    setInternalEditEvents((prev) => [...prev, event])
+    if (onEditEvent) {
+      onEditEvent(event)
+    }
+  }
+
   const { handleMouseDown, isDragging, activeEditEvent } = useComponentDragging(
     {
-      onEditEvent,
+      onEditEvent: handleEditEvent,
       cancelDrag,
       realToSvgProjection,
       svgToScreenProjection,
       circuitJson,
-      editEvents,
+      editEvents: [...editEvents, ...internalEditEvents],
```

Comment:
use a useMemo for this, you can call this intermediate array `editEventsWithUnappliedEvents`

Reply:
without useMemo this could cause a lot of rerenders